crates/dbsp/src/storage/buffer_cache/tests.rs

```diff
@@ -1,148 +0,0 @@
-use proptest::{proptest, test_runner::Config};
-use proptest_state_machine::{prop_state_machine, ReferenceStateMachine, StateMachineTest};
-use std::sync::Arc;
-use tempfile::TempDir;
-
-use crate::storage::{
-    backend::{
-        monoio_impl::MonoioBackend,
-        tests::{InMemoryBackend, Transition, MAX_TRANSITIONS},
-        FileHandle, ImmutableFileHandle, StorageControl, StorageExecutor, StorageRead,
-        StorageWrite,
-    },
-    buffer_cache::{BufferCache, TinyLfuCache},
-    test::init_test_logger,
-};
-
-#[monoio::test]
-#[should_panic]
-async fn overlaps_test() {
-    use super::FBuf;
-    use crate::storage::backend::tests::InMemoryBackend;
-
-    let cache = BufferCache::with_backend_lfu(
-        InMemoryBackend::<true>::default(),
-        Arc::new(TinyLfuCache::default()),
-    );
-    let fd = cache.create().await.unwrap();
-    let _fd2 = cache.create().await.unwrap();
-
-    let mut b1 = FBuf::with_capacity(2048);
-    b1.extend_from_slice(&vec![b'a'; 2048]);
-
-    cache.write_block(&fd, 1024, b1).await.unwrap();
-
-    let mut b2 = FBuf::with_capacity(1024);
-    b2.extend_from_slice(&vec![b'a'; 1024]);
-
-    cache.write_block(&fd, 512, b2).await.unwrap();
-}
-
-// Set up the state machine test using the `prop_state_machine!` macro
-prop_state_machine! {
-    #![proptest_config(Config {
-        verbose: 1,
-        .. Config::default()
-    })]
-
-    #[test]
-    fn cached_monoio_behaves_like_model(
-        sequential
-        1..MAX_TRANSITIONS
-        =>
-        BufferCache<MonoioBackend>
-    );
-}
-
-pub struct BufferCacheTest {
-    backend: BufferCache<MonoioBackend>,
-    _tmpdir: TempDir,
-}
-
-impl StateMachineTest for BufferCache<MonoioBackend> {
```

Comment:
does this no longer hold?

Reply:
It's because I changed the BufferCache so that it no longer implements the storage trait itself. I *think* I could restore that property if necessary.